code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -761,3 +761,79 @@ MATCHBOXES ARE ALSO IN FANCY.DM
 		lit = 0
 		update_brightness()
 		visible_message("<span class='warning'>Without warning, the flame on \the [src] suddenly goes out in a weak fashion.</span>")
+
+/obj/item/weapon/vape
+	name = "E-Cigarette"
+	desc = "More commony refered to as a mouth fedora"
+	icon = 'icons/obj/items.dmi'
+	icon_state = "vape"
+	var/datum/effect/effect/system/smoke_spread/smoke = new()
+	var/max_beaker_volume = 50
+	var/beaker = null
+	var/vape_amount = 5
+	var/emagged = FALSE
+
+/obj/item/weapon/vape/attack_self(mob/living/user)
+	if(emagged == TRUE)
+		playsound(get_turf(src), 'sound/effects/Explosion_Small1.ogg', 50, 0)
+		user.apply_damage(20, BURN, "head")
+		qdel(src)
+		var/datum/effect/effect/system/spark_spread/sparks = new /datum/effect/effect/system/spark_spread()
+		sparks.set_up(3, 0, get_turf(user))
+		sparks.start()
+		to_chat(user, "<span class='userdanger'>The [name] suddenly explodes in your mouth!</span>")
+	else
+
+		var/obj/item/weapon/reagent_containers/glass/beaker/B = beaker
+		user.delayNextAttack(50)
```

Comment:
Have you tested this? I'm not certain that `delayNextAttack()` delays `attack_self()` calls.